From de766256866fe7efceadcf0cf1bbf956c4eb0789 Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Thu, 4 Feb 2016 09:57:18 +0800
Subject: [PATCH 09/13] i2c: designware: check return value of kzalloc

Pointer 'buffer' returned from call to function 'kzalloc'
may be NULL and may be dereferenced.

Change-Id: I374d5cc82c602aa05928307c7a3855518660b998
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18604
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
Reviewed-on: https://android.intel.com:443/466283
---
 drivers/i2c/busses/i2c-designware-platdrv.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-designware-platdrv.c b/drivers/i2c/busses/i2c-designware-platdrv.c
index 7feb0ff..2278db99 100644
--- a/drivers/i2c/busses/i2c-designware-platdrv.c
+++ b/drivers/i2c/busses/i2c-designware-platdrv.c
@@ -134,7 +134,12 @@ dw_i2c_acpi_space_handler(u32 function, acpi_physical_address address,
 	function &= ACPI_IO_MASK; 
 	if (function == ACPI_READ) {
 		buffer = kzalloc(length, GFP_KERNEL);
-	
+		if (!buffer) {
+			pr_info("%s: buffer allocation failed!\n", __func__);
+			ACPI_FREE(ares);
+			return AE_NO_MEMORY;
+		}
+
 		msgs[0].addr = target;
 		msgs[0].flags = 0;
 		msgs[0].len = 1;
-- 
1.9.1

